src/compiler.py

```diff
@@ -1,25 +1,38 @@
 from typing import Optional
 import tempfile
 import subprocess
-
+import os
+import shutil
 from .helpers import try_remove
 
 
 class Compiler:
-    def __init__(self, compile_cmd: str, *, show_errors: bool) -> None:
+    def __init__(
+        self, compile_cmd: str, *, show_errors: bool, debug_mode: bool
+    ) -> None:
         self.compile_cmd = compile_cmd
         self.show_errors = show_errors
+        self.debug_mode = debug_mode
 
     def compile(self, source: str, *, show_errors: bool = False) -> Optional[str]:
         """Try to compile a piece of C code. Returns the filename of the resulting .o
         temp file if it succeeds."""
-        show_errors = show_errors or self.show_errors
+        show_errors = show_errors or self.show_errors or self.debug_mode
         with tempfile.NamedTemporaryFile(
             prefix="permuter", suffix=".c", mode="w", delete=False
         ) as f:
             c_name = f.name
             f.write(source)
 
+        if self.debug_mode:
+            debug_filepath = os.getcwd() + "/debug_source_" + c_name[-10:]
```

Comment:
`c_name[-10:]` is very ugly and quite possibly unportable. It's just to generate a random string, right? Can you do that in another way?

Also, could use `"."` instead of os.getcwd()

Reply:
you know I don't think it's even necessary to make a random string here